### PR TITLE
MM-38460 - Fix for: Delete entry confirmation dialog doesn't close

### PR DESCRIPTION
**Pull request by cpoile**
#### Summary
- The confirmation dialog would not close after confirming delete, so fixed that.
- That revealed a bigger problem: because we load the playbookRun in a parent component (it's not synced to the store), deleting the timeline event would not update the timeline. So we have to manually delete it at the level that the playbook is being loaded and stored. This means prop-drilling the delete function, which is not ideal. I think it's fine until we move to using the playbookRun in the store (which is a bigger change, out of scope for this bug fix PR).
- Did some refactoring cleanup.

#### Ticket Link
- https://mattermost.atlassian.net/browse/MM-38460

#### Checklist

- [ ] ~~Telemetry updated~~
- [ ] ~~Gated by experimental feature flag~~
- [ ] ~~Unit tests updated~~

### CODE REVIEW

**Review by lieut-data — APPROVED** [review]
Any chance of an e2e test to keep the modal working?

File: tests-e2e/cypress/integration/backstage/playbook_retro_timeline_spec.js [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+            teamId: testTeam.id,
+            playbookId: testPublicPlaybook.id,
+            playbookRunName: runName,
+            ownerUserId: testUser.id,
+        }).then((run) => {
+            runId = run.id;
+        });
+    });
+
+    beforeEach(() => {
```
Comment (lieut-data) [inline tests-e2e/cypress/integration/backstage/playbook_retro_timeline_spec.js:52]: Should we collapse the two `beforeEach` blocks?

**Review by lieut-data — APPROVED** [review]

**Comment by lieut-data** [comment]
Thanks for the e2e test, @cpoile :)

File: tests-e2e/cypress/integration/backstage/playbook_retro_timeline_spec.js (hunk 1 above)
Comment (cpoile) [reply tests-e2e/cypress/integration/backstage/playbook_retro_timeline_spec.js]: 🤦 Thanks.

**Review by crspeller — APPROVED** [review]

**Pull request merged**